### DISCUSSION TITLE
Installation de php 7.4 sur Uwamp

### DISCUSSION THREAD

**Discussion by loicbcn** [ddt-adl-sig/dev#7] [PHP]
Lorsque j'ai lancé php 7.2 ou 7.4 sur uwamp, curl refusait de fonctionner. C'est un problème fréquent, il y a pas mal de discussions à ce sujet sur stackoverflow notamment ... Mais peu se réfèrent à uwamp.
Voici la solution que j'ai trouvée et qui fonctionne sur mon poste.

### Installation d'une version plus récente d'Apache
Pour que php7 fonctionne correctement, notamment cUrl, il faut mettre à jour Apache:
* Récupérer Apache 2.4.49 Win32 ici: https://www.apachelounge.com/download/ (https://www.apachelounge.com/download/VS16/binaries/httpd-2.4.49-win32-VS16.zip)
* Dans UWAMP/bin, renommer le dossier "apache" en "__apache", par exemple (pour pouvoir revenir à l'ancienne version au cas où, et pour copier quelques fichiers)
* Dans UWAMP/bin toujours, créer un dossier "apache", et dézipper le contenu du dossier Apache24 contenu dans le fichier httpd-2.4.49-win32-VS16.zip

* Aller dans le répertoire "__apache/conf" et copier le fichier httpd_uwamp.conf, puis coller ce fichier dans "apache/conf"
* Dans httpd_uwamp.conf, vers la 3ème ligne du fichier, remplacer la ligne 
ServerRoot "{APACHEPATH}/"
par ces 2 lignes: 
Define SRVROOT "{APACHEPATH}/"
ServerRoot "${SRVROOT}"
Pour éviter le message d'erreur "Config variable ${SRVROOT} is not defined"

* Aller dans le répertoire "__apache" et copier le dossier "certificats", puis coller ce répertoire dans "apache"

### Installation de php
Ensuite, pour installer php7.4.24:
à cette url: https://windows.php.net/download#php-7.4 Télécharger le fichier zip correspondant à: VC15 x86 Thread Safe (Attention x86 et pas x64)
https://windows.php.net/downloads/releases/php-7.4.24-Win32-vc15-x86.zip

* Dans UWAMP/bin/php créer un répertoire php-7.4.24, puis dézipper dedans le zip php récupéré.

En démarrant UWAMP, la nouvelle version de php apparaît dans la liste des versions disponibles. 
A noter qu'après la mise à jour d'Apache, la version 5.6 de php continue de fonctionner.

## Installer Twig pour le projet, en utilisant composer sans l'installer.
* Télécharger le fichier composer.phar ici: https://getcomposer.org/download/ (Manual download - latest stable)

* Placer composer.phar à la racine du projet. Par exemple (uwamp/www/votreprojet/)

* Créer le fichier composer.json à la racine du projet et coller dedans ce json:
{
 "require": {
    "twig/twig": "*"
  }
}

* Dans l'interface de uwamp (bouton phpconfig à côté de la version de php), vérifier que la version php est celle du projet
* Activer les extensions openssl, et curl si elles ne le sont pas déjà.

* Copier le php.ini situé dans UWAMP/bin/apache dans UwAmp/bin/php/php-7.4.24
En effet, dans uwamp, le dossier php contient par défaut php_uwamp.ini, qui est interprété par le serveur pour créer le php.ini dans le dossier apache.
Le fichier php.ini ainsi placé dans le dossier php permet à php, lorsque l'on lance une commande php depuis la ligne de commande, de retrouver sa configuration 
(notamment les extensions à charger)

* Lancer une ligne de commande dans le répertoire du projet et taper:
php composer.phar install

Si tout se passe bien, twig et ses dépendances sont téléchargés dans le répertoire du projet.

**Comment by PhilippeD31** [comment]
Merci Loïc !  
Apparemment, pour faire marcher phpMyAdmin, il faudra aussi créer un dossier **sessions** dans UWAMP/bin/apache (avec les droits d'écriture).
